MapboxCoreNavigation/MBRouteController.m

```diff
@@ -32,5 +33,13 @@ - (NSString * _Nonnull)md5 {
     
     return  output;
 }
+
 @end
 
+@implementation MBXAccounts (MBXDefault)
+
++ (void)load {
```

Comment:
Making the activation automatic on application launch is an improvement, to be sure: it streamlines the setup experience and removes an error-prone step (where an error takes a long time to discover). But it also removes the developer’s ability to decide when to activate a trip, which may be relevant for an application that puts turn-by-turn navigation behind an in-app purchase or something to that effect.

The fundamental challenge is that the typical workflow requires the developer to use MapboxDirections.swift before involving the navigation SDK in any way, yet that initial MapboxDirections.swift usage also needs to be accounted for in a trip. It makes me wonder whether we could impose an expectation that NavigationRouteOptions or NavigationMatchOptions be used in order to begin a “trip”, as opposed to RouteOptions or MatchOptions. These subclasses are part of the navigation SDK, so we could activate the navigation SKU ID as soon as one is initialized.

/cc @coxchapman

Reply:
> But it also removes the developer’s ability to decide when to activate a trip

Activating a sku identifier only set the identifier. The first service library (Speech or Directions) that makes a subsequent request, will generate the first token and the countdown towards token rotation begins.